upf/upf_cli.c

```diff
@@ -254,7 +254,15 @@ upf_ueip_pool_add_del_command_fn (vlib_main_t * vm,
   nwi_name = upf_name_to_labels (nwi_s);
   vec_free (nwi_s);
 
-  rv = vnet_upf_ue_ip_pool_add_del (name, nwi_name, is_add);
+  if (vec_len (nwi_name) > 64)
+    return clib_error_return (0, "NWI name(encoded) has to fit in 64 bytes");
+  if (vec_len (identity) > 64)
+    return clib_error_return (0, "UE IP pool name has to fit in 64 bytes");
```

Comment:
In this case, we need to free these vectors. Also, there's a similar problem with `nwi_s` which existed in the original code. One way to fix it: assign `0` to `name`, `nwi_name`, and `identity` at the beginning of the func, assign `error` val and `goto done` instead of returning immediately upon an error, and at the `done` label, `vec_free()` them. Note that you do not need to check if the vars are `0` or not, as `vec_free()` does the check